src/type.ml

```diff
@@ -14,11 +14,42 @@ type t =
   | ForallTyps of Name.t list * t
   | FunTyps of Name.t list * t
   | Error of string
-
 and arity_or_typ =
   | Arity of int
   | Typ of t
 
+
+let to_string : t -> string = function
+  | Variable _ -> "Var"
+  | Arrow _ -> "Arrow"
+  | Sum _ -> "Sum"
+  | Tuple _ -> "Tuple"
+  | Apply (mpath, _) -> MixedPath.to_string mpath
+  | Package _ -> "Package"
+  | ForallModule _ -> "ForallModule"
+  | ForallTyps _ -> "ForallTyps"
+  | FunTyps _ -> "FunTyps"
+  | Error s -> "Error" ^ s
+
+let equal (t1 : t) (t2 : t): bool =
```

Comment:
As this function is not the true equality, I would use another name than `equal` or put this function and `to_string` and `compare` in a sub-module saying that this is related to the tag name generation.